Sources/GDSAnalytics/Events/Buttons/ButtonEvent.swift

```diff
@@ -11,11 +11,19 @@ public struct ButtonEvent: Event {
         ].mapValues(\.formattedAsParameter)
     }
     
-    public init(textKey: String, _ variableKeys: String...) {
-        self.init(textKey: textKey, variableKeys: variableKeys)
+    public init(eventName: EventName = .navigation,
```

Comment:
wary that this means a `formResponse` eventName could be used for a `ButtonEvent` which doesn't seem right, do we need to split event name up?

Or alternatively, should not be creating different `Event` implementations for `popup`, etc.

Reply:
Would be good to chat about this, I would think a more flexible event would be nice but maybe we just do a concrete type for each as you say, eg popup

Reply:
Understood now, and updated with the relevant event types!